framework/Models/PostProcessors/PostProcessorFunctions/HS2PS.py

```diff
@@ -0,0 +1,202 @@
+
+  def run(self,inputIn):
+    """
+    This method performs the actual transformation of the data object from history set to point set
+      @ In, inputIn, dict, dictionaries which contains the data inside the input DataObjects
+        inputIn = {'Data':listData, 'Files':listOfFiles},
+        listData has the following format: (listOfInputVars, listOfOutVars, DataDict) with
+        DataDict is a dictionary that has the format
+            dataDict['dims']     = dict {varName:independentDimensions}
+            dataDict['metadata'] = dict {metaVarName:metaVarValue}
+            dataDict['type'] = str TypeOfDataObject
+            dataDict['inpVars'] = list of input variables
+            dataDict['outVars'] = list of output variables
+            dataDict['numberRealization'] = int SizeOfDataObject
+            dataDict['name'] = str DataObjectName
+            dataDict['metaKeys'] = list of meta variables
+            dataDict['data'] = dict {varName: varValue(1-D or 2-D numpy array)}
+      @ Out, outputDic, dict, output dictionary contains the converted PointSet data
+    """
+    #### convert to use Dataset, the following has been tested. However, this changes
+    #### will cause the failure of DataMining since interfacePP is used by DataMining
+    #### for pre-processing data. When we convert all interfacePP to use Dataset, we can
+    #### use the following code, and eventually remove the dict option
+    #      @ In, inputIn, dict, dictionary of data.
+    #          inputIn = {'Data':listData, 'Files':listOfFiles},
+    #          listData has the following format: (listOfInputVars, listOfOutVars, xr.Dataset)
+    #      @ Out, outDataset, xarray.Dataset, output dataset
+    # inpVars, outVars, data = inputIn['Data'][0]
+    # if self.features == 'all':
+    #   self.features = outVars
+    # outDataset = data.drop_dims(self.pivotParameter)
+    # featDataset = data[self.features]
+    # if featDataset[self.features[-1]].isnull().sum() > 0:
+    #   self.raiseAnError(IOError, 'Found misalignment in provided DataObject!')
+    # numRlz = data.dims['RAVEN_sample_ID']
+    # featData = featDataset.to_array().values.transpose(1, 0, 2).reshape(numRlz, -1)
+    # varNames = [str(i) for i in range(featData.shape[-1])]
+    # convertedFeat = xr.DataArray(featData, dims=('RAVEN_sample_ID', 'outVars'), coords={'RAVEN_sample_ID':data['RAVEN_sample_ID'], 'outVars':varNames})
+    # convertedFeatDataset = convertedFeat.to_dataset(dim='outVars')
+    # outDataset = xr.merge([outDataset, convertedFeatDataset])
+    # ## self.transformationSettings is used by _inverse method when doing DataMining
+    # self.transformationSettings['vars'] = copy.deepcopy(self.features)
+    # self.transformationSettings['timeLength'] = data[self.pivotParameter].size
+    # self.transformationSettings['timeAxis'] = data[self.pivotParameter][0]
+    # self.transformationSettings['dimID'] = list(outDataset.keys())
+    # return outDataset
+    #########
```

Comment:
DataMining are using several interfaced postprocessors as pre-processors. In order to make the DataMining test green, we need to convert these post processors to use Dataset at the same time. I have reverted the changes here, and I am proposing to work on this issue in next pull request, since there is already a lot of file changes in PR, and the conversion will take longer time than I thought.